src/core/net/ip6.cpp

```diff
@@ -1490,6 +1490,25 @@ const char *Ip6::IpProtoToString(uint8_t aIpProto)
     return Stringify::Lookup(aIpProto, kIpProtoTable, "Unknown");
 }
 
+#if OT_SHOULD_LOG_AT(OT_LOG_LEVEL_NOTE)
```

Comment:
It really depends on the actually log level used for the log code.

Should we just leave this method unclosed with `OT_SHOULD_LOG_AT` ? 

I think unused code is automatically removed after compiling anyway.

Reply:
I removed the check in new push.

I recall some older toolchains may not remove `static` constant definitions in an unused function/method. But I think/expect most toolchains (practically all toolchain that may be used for building OT stack) to correctly take care of it (i.e., remove unused definitions). I agree it may be better to not tie `EcnToString()` to any log level.